packages/knip/src/plugins/next/index.ts

```diff
@@ -9,28 +11,40 @@ const enablers = ['next'];
 
 const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
 
-const entry = ['next.config.{js,ts,cjs,mjs}'];
+const config = ['next.config.{js,ts,cjs,mjs}'];
 
-const productionEntryFilePatternsWithoutSrc = [
+const defaultPageExtensions = ['{js,jsx,ts,tsx}'];
+
+const productionEntryFilePatterns = [
   '{instrumentation,middleware}.{js,ts}',
   'app/global-error.{js,jsx,ts,tsx}',
   'app/**/{error,layout,loading,not-found,page,template,default}.{js,jsx,ts,tsx}',
   'app/**/route.{js,jsx,ts,tsx}',
   'app/{manifest,sitemap,robots}.{js,ts}',
   'app/**/{icon,apple-icon}.{js,jsx,ts,tsx}',
   'app/**/{opengraph,twitter}-image.{js,jsx,ts,tsx}',
-  'pages/**/*.{js,jsx,ts,tsx}',
 ];
 
-const production = [
-  ...productionEntryFilePatternsWithoutSrc,
-  ...productionEntryFilePatternsWithoutSrc.map(pattern => `src/${pattern}`),
-];
+const getEntryFilePatterns = (pageExtensions = defaultPageExtensions) => {
+  const pages = pageExtensions.map(ext => `pages/**/*.${ext}`);
+  const patterns = [...productionEntryFilePatterns, ...pages];
+  return [...patterns, ...patterns.map(pattern => `src/${pattern}`)];
```

Comment:
same here, wondering about if duplicates could be a problem

Reply:
This is all local, I don't think there's a huge risk of dupes.